crates/utils/src/utils/validation.rs

```diff
@@ -348,6 +349,45 @@ pub fn build_url_str_without_scheme(url_str: &str) -> LemmyResult<String> {
   Ok(out)
 }
 
+// Shorten a string to n chars, being mindful of unicode grapheme
+// boundaries
+#[allow(clippy::indexing_slicing)]
+pub fn truncate_for_db(string: String, len: usize) -> String {
+  if string.chars().count() <= len {
+    return string;
+  }
+
+  let offset = string
+    .char_indices()
+    .nth(len)
+    .unwrap_or(string.char_indices().last().unwrap_or_default());
+  let graphemes: Vec<(usize, _)> = string.as_str().grapheme_indices(true).collect();
+  let mut index = 0;
+  // Walk the string backwards and find the first char within our length
+  for idx in (0..graphemes.len()).rev() {
+    let grapheme = graphemes[idx];
+    if grapheme.0 < offset.0 {
+      index = idx;
+      break;
+    }
+  }
+  let grapheme = graphemes[index];
+  let grapheme_count = grapheme.1.chars().count();
+  // `take` isn't inclusive, so if the last grapheme can fit we add its char
+  // length
+  let char_count = if grapheme_count + grapheme.0 <= len {
+    grapheme.0 + grapheme_count
+  } else {
+    grapheme.0
+  };
+
+  string.chars().take(char_count).collect::<String>()
+}
+
+pub fn truncate_description(string: String) -> String {
+  truncate_for_db(string, SITE_DESCRIPTION_MAX_LENGTH)
+}
```

Comment:
I wonder if there is any way to get the `max_length` attributes from [schema.rs](https://github.com/LemmyNet/lemmy/blob/main/crates/db_schema/src/schema.rs#L186), then we dont have to define them twice.

Reply:
Pry not worth it anyway.